.github/workflows/lint.yaml

```diff
@@ -0,0 +1,122 @@
+name: Linter
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+# cancel the in-progress workflow when PR is refreshed.
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event_name == 'pull_request' && github.head_ref || github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  codespell:
+    name: codespell
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: codespell
+        uses: codespell-project/actions-codespell@master
+        with:
+          # LICENSE: skip file because codespell wants to flag complies, which we may want to flag
+          # in other places, so ignore the file itself assuming it is correct
+          # crds.yaml, resources.yaml: CRD files are fully generated from content we control (should
+          # be flagged elsewhere) and content we don't control (can't fix easily), so ignore
+          skip: .git,*.png,*.jpg,*.svg,*.sum
+          # aks: Amazon Kubernetes Service
+          # keyserver: flag to apt-key
+          # atleast: codespell wants to flag any 'AtLeast' method
+          # ser, ist: MIME types in pkg/operator/ceph/object/mime.go
+          # ba: daemon ID test in pkg/operator/k8sutil/name_test.go
+          # iam: Identity and Access Management, e.g., AWS IAM
+          # te: udev persistent naming test in pkg/daemon/ceph/osd/daemon_test.go
+          # parm: modinfo parameter
+          # assigment: inherited from K8s TopologySpreadConstraints dependency
+          # ro, RO: means read-only
+          ignore_words_list: aks,keyserver,atleast,ser,ist,ba,iam,te,parm,assigment,ro,RO,addin,NotIn
+          check_filenames: true
+          check_hidden: true
+  misspell:
+    name: misspell
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: misspell
+        uses: reviewdog/action-misspell@v1
+
+  golangci:
+    name: golangci-lint
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: actions/setup-go@v5
+        with:
+          go-version: "1.21"
+      - name: golangci-lint
+        uses: golangci/golangci-lint-action@v6
+        with:
+          # Required: the version of golangci-lint is required and must be specified without patch version: we always use the latest patch version.
+          version: v1.55
+
+          # Optional: working directory, useful for monorepos
+          # working-directory: somedir
+
+          # Optional: golangci-lint command line arguments.
+          args: --config=.golangci.yml
```

Comment:
`.golangci.yml` config file is missing?

Reply:
it is already present https://github.com/ceph/ceph-csi-operator/blob/main/.golangci.yml